From 28ae8064096488ca89bd3e308a57bb4dd1d491e1 Mon Sep 17 00:00:00 2001
From: Edsko de Vries <edsko@well-typed.com>
Date: Wed, 8 May 2013 12:22:22 +0100
Subject: [PATCH] Backport patch to #7231 (commit 772e6d2)

---
 compiler/main/GhcMake.hs | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/compiler/main/GhcMake.hs b/compiler/main/GhcMake.hs
index 091e1be..3c90c1a 100644
--- a/compiler/main/GhcMake.hs
+++ b/compiler/main/GhcMake.hs
@@ -252,11 +252,18 @@ load2 how_much mod_graph = do
             stable_mg = 
                 [ AcyclicSCC ms
                 | AcyclicSCC ms <- full_mg,
-                  ms_mod_name ms `elem` stable_obj++stable_bco,
-                  ms_mod_name ms `notElem` [ ms_mod_name ms' | 
-                                                AcyclicSCC ms' <- partial_mg ] ]
-
-            mg = stable_mg ++ partial_mg
+                  ms_mod_name ms `elem` stable_obj++stable_bco ]
+
+            -- the modules from partial_mg that are not also stable
+            -- NB. also keep cycles, we need to emit an error message later
+            unstable_mg = filter not_stable partial_mg
+              where not_stable (CyclicSCC _) = True
+                    not_stable (AcyclicSCC ms)
+                       = ms_mod_name ms `notElem` stable_obj++stable_bco
+
+            -- Load all the stable modules first, before attempting to load
+            -- an unstable module (#7231).
+            mg = stable_mg ++ unstable_mg
 
         -- clean up between compilations
         let cleanup hsc_env = intermediateCleanTempFiles dflags
-- 
1.7.12.4 (Apple Git-37)

