diff -ur orig/blaze-builder-0.3.1.0/Blaze/ByteString/Builder/ByteString.hs blaze-builder-0.3.1.0/Blaze/ByteString/Builder/ByteString.hs
--- orig/blaze-builder-0.3.1.0/Blaze/ByteString/Builder/ByteString.hs	2013-03-06 11:25:03.000000000 +0000
+++ blaze-builder-0.3.1.0/Blaze/ByteString/Builder/ByteString.hs	2013-03-06 11:24:42.000000000 +0000
@@ -38,6 +38,7 @@
 import qualified Blaze.ByteString.Builder.Internal as I   (insertByteString)
 
 import Foreign
+import GHC.ForeignPtr (unsafeForeignPtrToPtr)
 import Data.Monoid
 
 import qualified Data.ByteString      as S
diff -ur orig/blaze-builder-0.3.1.0/Blaze/ByteString/Builder/Internal/Buffer.hs blaze-builder-0.3.1.0/Blaze/ByteString/Builder/Internal/Buffer.hs
--- orig/blaze-builder-0.3.1.0/Blaze/ByteString/Builder/Internal/Buffer.hs	2013-03-06 11:25:03.000000000 +0000
+++ blaze-builder-0.3.1.0/Blaze/ByteString/Builder/Internal/Buffer.hs	2013-03-06 11:24:15.000000000 +0000
@@ -42,6 +42,7 @@
 
 import Prelude
 import Foreign
+import GHC.ForeignPtr (unsafeForeignPtrToPtr)
 
 import qualified Data.ByteString      as S
 
diff -ur orig/blaze-builder-0.3.1.0/Blaze/ByteString/Builder/Internal.hs blaze-builder-0.3.1.0/Blaze/ByteString/Builder/Internal.hs
--- orig/blaze-builder-0.3.1.0/Blaze/ByteString/Builder/Internal.hs	2013-03-06 11:25:03.000000000 +0000
+++ blaze-builder-0.3.1.0/Blaze/ByteString/Builder/Internal.hs	2013-03-06 11:24:28.000000000 +0000
@@ -50,6 +50,7 @@
 ) where
 
 import Foreign
+import GHC.ForeignPtr (unsafeForeignPtrToPtr)
 
 import Control.Monad (unless)
 
