diff -ur orig/filemanip-0.3.6.2/filemanip.cabal filemanip-0.3.6.2/filemanip.cabal
--- orig/filemanip-0.3.6.2/filemanip.cabal	2013-03-06 11:32:49.000000000 +0000
+++ filemanip-0.3.6.2/filemanip.cabal	2013-03-06 11:29:19.000000000 +0000
@@ -17,7 +17,7 @@
 Extra-Source-Files: README.markdown
 
 Library
-  build-depends: base < 5, bytestring, directory, filepath, mtl, unix-compat
+  build-depends: base < 5, bytestring, directory, filepath, mtl, unix-compat >= 0.3
   if impl(ghc >= 6.10)
     build-depends:
       base >= 4
