diff -ur orig/HUnit-1.2.5.1/Test/HUnit/Lang.hs HUnit-1.2.5.1/Test/HUnit/Lang.hs
--- orig/HUnit-1.2.5.1/Test/HUnit/Lang.hs	2013-03-06 16:13:36.000000000 +0000
+++ HUnit-1.2.5.1/Test/HUnit/Lang.hs	2013-03-06 16:12:48.000000000 +0000
@@ -1,6 +1,6 @@
 -- | This module abstracts the differences between implementations of 
 -- Haskell (e.g., GHC, Hugs, and NHC).
-
+{-# LANGUAGE StandaloneDeriving, DeriveDataTypeable #-}
 module Test.HUnit.Lang
 (
   Assertion,
@@ -84,9 +84,8 @@
 hunitFailureTc = mkTyCon "HUnitFailure"
 #endif
 {-# NOINLINE hunitFailureTc #-}
- 
-instance Typeable HUnitFailure where
-    typeOf _ = mkTyConApp hunitFailureTc []
+
+deriving instance Typeable HUnitFailure
 #ifdef BASE4
 instance Exception HUnitFailure
 
