diff -ur orig/test-framework-0.8/Test/Framework/Runners/Console.hs test-framework-0.8/Test/Framework/Runners/Console.hs
--- orig/test-framework-0.8/Test/Framework/Runners/Console.hs	2013-03-06 16:08:23.000000000 +0000
+++ test-framework-0.8/Test/Framework/Runners/Console.hs	2013-03-06 16:08:06.000000000 +0000
@@ -23,14 +23,6 @@
 import Data.Monoid
 
 
-instance Functor OptDescr where
-    fmap f (Option a b arg_descr c) = Option a b (fmap f arg_descr) c
-
-instance Functor ArgDescr where
-    fmap f (NoArg a) = NoArg (f a)
-    fmap f (ReqArg g s) = ReqArg (f . g) s
-    fmap f (OptArg g s) = OptArg (f . g) s
-
 -- | @Nothing@ signifies that usage information should be displayed.
 -- @Just@ simply gives us the contribution to overall options by the command line option.
 type SuppliedRunnerOptions = Maybe RunnerOptions
