diff -ur orig/unix-compat-0.4.1.0/unix-compat.cabal unix-compat-0.4.1.0/unix-compat.cabal
--- orig/unix-compat-0.4.1.0/unix-compat.cabal	2013-03-06 11:31:20.000000000 +0000
+++ unix-compat-0.4.1.0/unix-compat.cabal	2013-03-06 11:30:56.000000000 +0000
@@ -60,7 +60,7 @@
       System.PosixCompat.Internal.Time
 
   else
-    build-depends: unix >= 2.4 && < 2.7
+    build-depends: unix >= 2.4 && < 2.8
     include-dirs: include
     includes: HsUnixCompat.h
     install-includes: HsUnixCompat.h
