diff -ur orig/unordered-containers-0.2.3.0/Data/HashMap/Array.hs unordered-containers-0.2.3.0/Data/HashMap/Array.hs
--- orig/unordered-containers-0.2.3.0/Data/HashMap/Array.hs	2013-03-06 13:18:57.000000000 +0000
+++ unordered-containers-0.2.3.0/Data/HashMap/Array.hs	2013-03-06 13:17:09.000000000 +0000
@@ -52,7 +52,7 @@
 import Control.Applicative (Applicative)
 import Control.DeepSeq
 import Control.Monad.ST hiding (runST)
-import GHC.Exts
+import GHC.Exts hiding (toList, fromList)
 import GHC.ST (ST(..))
 import Prelude hiding (filter, foldr, length, map, read)
 import qualified Prelude
